types/react-native-awesome-card-io/index.d.ts

```diff
@@ -0,0 +1,74 @@
+    scanInstructions?: string; // - Set the scan instruction text. If nil, use the default text.
+    scanExpiry?: boolean; // true - Set to false if you don't want the camera to try to scan the card expiration.
+
+    // iOS only
+    detectionMode?: CardIODetectionMode; // false - Set the detection mode. (iOS)
+    scannedImageDuration?: number; // 0.1 - How long card.io will display an image of the card with the computed card number superimposed after a successful scan.
+    allowFreelyRotatingCardGuide?: boolean; // true - By default, in camera view the card guide and the buttons always rotate to match the device's orientation.
+}
+
+export interface CardIOViewProps extends CardIOCommonProps {
+    didScanCard: (card: CardDetails) => void; // This function will be called when the CardIOView completes its work and returns a CreditCard.
+    style?: StyleProp<ViewStyle>; // Style props of CardIOViewProps
+}
+
+export interface CardIOModuleProps extends CardIOCommonProps {
+    suppressManualEntry?: boolean; // false - Set to true to prevent card.io from showing its "Enter Manually" button.
+    suppressConfirmation?: boolean; // false - If true, don't have the user confirm the scanned card, just return the results immediately.
+    requireExpiry?: boolean; // true - Set to false if you don't need to collect the card expiration.
+    requireCVV?: boolean; // true - Set to false if you don't need to collect the CVV from the user.
+    requirePostalCode?: boolean; // false - Set to false if you need to collect the billing postal code.
+    restrictPostalCodeToNumericOnly?: boolean; // false - Set to true if the postal code should only collect numeric input.
+    requireCardholderName?: boolean; // false - Set to true if you need to collect the cardholder name.
+
+    // iOS Only
+    disableBlurWhenBackgrounding?: boolean; // false - Disable the blur of the screen when the app is backgrounded.
+    keepStatusBarStyle?: boolean; // false - If true, the status bar's style will be kept as whatever your app has set it to.
+    suppressScannedCardImage?: boolean; // false - If true, instead of displaying the image of the scanned card, present the manual entry screen with the scanned card number prefilled.
+
+    // Android Only
+    noCamera?: boolean; // false If set, the card will not be scanned with the camera.
+    unblurDigits?: number; // -1 Privacy feature. How many of the Card number digits NOT to blur on the resulting image. Setting it to 4 will blur all digits except the last four.
+    usePaypalActionbarIcon?: boolean; //  = false; Use the PayPal icon in the ActionBar.
+}
+
+export type CardIODetectionMode = 'IMAGE_AND_NUMBER' | 'IMAGE' | 'AUTOMATIC';
+
+export const CardIOUtilities: {
+    preload: () => void; // iOS only - prepares card.io to launch faster.
+    CAN_READ_CARD_WITH_CAMERA: boolean;
+    DETECTION_MODE: CardIODetectionMode;
+};
+
+export const CardIOModule: {
```

Comment:
It would be cleaner to do the following:
```ts
export namespace CardIOUtilities {
  export function scanCard(config?: CardIOModuleProps): Promise<CardDetails>;
}
```

Reply:
Hi @uniqueiniquity, I've updated my PR with requested changes.